mypyc/codegen/emitfunc.py

```diff
@@ -471,6 +471,15 @@ def visit_load_mem(self, op: LoadMem) -> None:
         type = self.ctype(op.type)
         self.emit_line('%s = *(%s *)%s;' % (dest, type, src))
 
+    def visit_get_element_ptr(self, op: GetElementPtr) -> None:
+        dest = self.reg(op)
+        src = self.reg(op.src)
+        # TODO: support tuple type
+        assert isinstance(op.src_type, RStruct)
+        assert op.index < len(op.src_type.offsets), "Invalid element index."
+        offset = op.src_type.offsets[op.index]
+        self.emit_line('%s = (char *)%s + %s;' % (dest, src, offset))
```

Comment:
Generate `<dest> = &<src>.<field_name>` instead (by default). This produces more readable C. 

We'll use the offsets in non-C backends. It's still important that we can calculate the offsets, but not for the C back end. 

To test the offset calculation for real, it would be good to have some configuration option to use pointer arithmetic in tests, even when using the C back end. You can do this in a separate PR to keep this one simpler.